File: CSharp/DP.Domain/Samples/Mediator/MediatorAverage.cs
using System.Diagnostics;

namespace DP.Domain.Samples.Mediator
{
    public class MediatorAverage : IMediator
    {
        public IColleague Option { get; set; }
        public IColleague Credit { get; set; }
        public IColleague Loan { get; set; }

        public MediatorAverage()
        {
            this.Option = new OptionColleague();
            this.Credit = new CreditColleague();
            this.Loan = new LoanColleague();
        }
        public decimal Score()
        {
            decimal scoreOption = this.Option.Score();
            decimal scoreCredit = this.Credit.Score();
            decimal scoreLoan = this.Loan.Score();
            return  (scoreOption + scoreCredit + scoreLoan) / 3;
        }
    }
}

File: Python/Samples/Mediator/MediatorAverage.py
from Mediator import Mediator
from OptionColleague import OptionColleague
from CreditColleague import CreditColleague
from LoanColleague import LoanColleague

class MediatorAverage(Mediator):

    def __init__(self):
        self.option = OptionColleague()
        self.credit = CreditColleague()
        self.loan = LoanColleague()

    def score(self):
        scoreOption = self.option.score()
        scoreCredit = self.credit.score()
        scoreLoan = self.loan.Score()

        return (scoreOption + scoreCredit + scoreLoan) / 3
